Client/Frontend/Browser/BrowserViewController.swift

```diff
@@ -696,21 +696,26 @@ extension BrowserViewController: URLBarDelegate {
         if pasteboardContents != nil {
             let pasteAndGoAction = UIAlertAction(title: NSLocalizedString("Paste & Go", comment: "Paste the URL into the location bar and visit"), style: .Default, handler: { (alert: UIAlertAction!) -> Void in
                 self.urlBar(urlBar, didSubmitText: pasteboardContents!)
+                completion?(success: true)
             })
             longPressAlertController.addAction(pasteAndGoAction)
 
             let pasteAction = UIAlertAction(title: NSLocalizedString("Paste", comment: "Paste the URL into the location bar"), style: .Default, handler: { (alert: UIAlertAction!) -> Void in
                 urlBar.updateURLBarText(pasteboardContents!)
+                completion?(success: true)
             })
             longPressAlertController.addAction(pasteAction)
         }
 
         let copyAddressAction = UIAlertAction(title: NSLocalizedString("Copy Address", comment: "Copy the URL from the location bar"), style: .Default, handler: { (alert: UIAlertAction!) -> Void in
             UIPasteboard.generalPasteboard().string = urlBar.currentURL?.absoluteString
+            completion?(success: true)
         })
         longPressAlertController.addAction(copyAddressAction)
 
-        let cancelAction = UIAlertAction(title: NSLocalizedString("Cancel", comment: "Cancel alert view"), style: .Cancel, handler: nil)
+        let cancelAction = UIAlertAction(title: NSLocalizedString("Cancel", comment: "Cancel alert view"), style: .Cancel, handler: { (alert: UIAlertAction!) -> Void in
+            completion?(success: true)
+        })
```

Comment:
Why is the completion called when one of the action sheet buttons is called? Doesn't all that happen after we have recoginized the long press? What needs to be completed here? I don't see why that needs to be a concern of this code.